organizationdomain/client.go

```diff
@@ -0,0 +1,122 @@
+// Package organizationdomain provides the Organization Domains API.
+package organizationdomain
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+)
+
+//go:generate go run ../cmd/gen/main.go
+
+const path = "/organizations"
+
+// Client is used to invoke the Organization Domains API.
+type Client struct {
+	Backend clerk.Backend
+}
+
+func NewClient(config *clerk.ClientConfig) *Client {
+	return &Client{
+		Backend: clerk.NewBackend(&config.BackendConfig),
+	}
+}
+
+type CreateParams struct {
+	clerk.APIParams
+	Name           string `json:"name" form:"name"`
+	OrganizationID string `json:"-" form:"-"`
+	EnrollmentMode string `json:"enrollment_mode" form:"enrollment_mode"`
+	Verified       *bool  `json:"verified" form:"verified"`
+}
+
+// Create adds a new domain to the organization.
+func (c *Client) Create(ctx context.Context, params *CreateParams) (*clerk.OrganizationDomain, error) {
+	path, err := clerk.JoinPath(path, params.OrganizationID, "/domains")
+	if err != nil {
+		return nil, err
+	}
+	req := clerk.NewAPIRequest(http.MethodPost, path)
+	req.SetParams(params)
+	domain := &clerk.OrganizationDomain{}
+	err = c.Backend.Call(ctx, req, domain)
+	return domain, err
+}
+
+type UpdateParams struct {
+	clerk.APIParams
+	OrganizationID string  `json:"-" form:"-"`
+	DomainID       string  `json:"-" form:"-"`
+	EnrollmentMode *string `json:"enrollment_mode" form:"enrollment_mode"`
+	Verified       *bool   `json:"verified" form:"verified"`
+}
+
+// Update updates an organization domain.
+func (c *Client) Update(ctx context.Context, params *UpdateParams) (*clerk.OrganizationDomain, error) {
```

Comment:
:wrench: The convention is to have Update methods accept the resource ID as a separate parameter. See here for an example: https://github.com/clerk/clerk-sdk-go/blob/807c12461675e18bb88c5162b99cd3951e803902/organization/client.go#L75